src/frontend/apps/desk/src/pages/mail-domains/[name].tsx

```diff
@@ -0,0 +1,66 @@
+import { Loader } from '@openfun/cunningham-react';
+import { useRouter as useNavigate } from 'next/navigation';
+import { useRouter } from 'next/router';
+import { ReactElement, useMemo } from 'react';
+
+import { Box } from '@/components';
+import { TextErrors } from '@/components/TextErrors';
+import {
+  EnumMailDomainsOrdering,
+  MailDomain as MailDomainType,
+  MailDomainsContent,
+  MailDomainsLayout,
+} from '@/features/mail-domains';
+import { useMailDomains } from '@/features/mail-domains/api/useMailDomains';
+import { NextPageWithLayout } from '@/types/next';
+
+const Page: NextPageWithLayout = () => {
+  const router = useRouter();
+
+  if (router?.query?.name && typeof router.query.name !== 'string') {
+    throw new Error('Invalid mail domain name');
+  }
+
+  const { name } = router.query;
+
+  const { data, isLoading, isError, error } = useMailDomains({
+    ordering: EnumMailDomainsOrdering.BY_CREATED_AT_DESC,
+  });
+
+  const navigate = useNavigate();
+
+  const mailDomains = useMemo(() => {
+    return data?.pages.reduce((acc, page) => {
+      return acc.concat(page.results);
+    }, [] as MailDomainType[]);
+  }, [data?.pages]);
+
+  const mailDomain = mailDomains?.find(
+    (mailDomain) => mailDomain.name === name,
+  );
+
+  if (error?.status === 404 || (!isLoading && !mailDomain)) {
```

Comment:
`mailDomain` is maybe not available because of an error 500, or other.
By doing like that everything will go to 404.

```suggestion
  if (error?.status === 404) {
```